changegen/__main__.py

```diff
@@ -103,6 +141,19 @@ def main(*args: tuple, **kwargs: dict):
     setup_logging(debug=kwargs["debug"])
     logging.debug(f"Args: {kwargs}")
 
+    # Check for ID collisions and warn
+    try:
```

Comment:
Nodes way and relations have different id spaces, so it's ok to have way ids the collide with node ids. But there is probably nothing to be gained by allowing that, even if it is valid.